tripy/tripy/flat_ir/ops/base.py

```diff
@@ -101,3 +134,75 @@ def _op_name(self) -> str:
             The name of this operation.
         """
         return self.__class__.__name__
+
+
+class FlatIRFunction(BaseProducer):
+    """
+    Represents a function in the Flat IR.
+
+    This class encapsulates a function with its inputs, outputs, and operations.
+    It inherits from BaseProducer, allowing it to be used in the IR graph.
+    """
+
+    def __init__(self, name: str, inputs: List["FlatIRTensor"], outputs: List["FlatIRTensor"]):
+        """
+        Initialize a FlatIRFunction.
+
+        Args:
+            name (str): The name of the function.
+            inputs (List[FlatIRTensor]): List of input tensors.
+            outputs (List[FlatIRTensor]): List of output tensors.
+        """
+        self.name = name
+        self.inputs = inputs
+        self.outputs = outputs
+        self.ops: List[BaseFlatIROp] = []
+        self.trace_input_names = [inp.name for inp in inputs]
+        self.trace_output_names = [out.name for out in outputs]
```

Comment:
So is the function boundary always aligned to trace ops?

Reply:
Yes, that is the idea as of now. Capturing all Flat IR build operations inside a given trace seemed simplest. Also, it will be consistent every time the trace op is invoked. 
TODO: The issue here is that we create a new function every time an op is traced because the trace can be different based on inputs IIUC.